sdks/java/core/src/main/java/org/apache/beam/sdk/util/IOChannelFactory.java

```diff
@@ -103,4 +103,9 @@
 
   /** Converts the given string to a {@link Path}. */
   Path toPath(String path);
+
+  /**
+   * Delete files with the given {@code uris} if they exist.
+   */
+  void deleteIfExists(Collection<String> uris) throws IOException;
```

Comment:
I think the actual problem is that you modified `IOChannelFactory` rather than `FileOperations` in `FileBasedSink`, which is the place this code actually belong. You are right that `IOChannelFactory` does not have any delete functionality.

That already has a `remove` function you can use. Perhaps as-is, perhaps with a try and catch.